sdks/java/io/jdbc/src/main/java/org/apache/beam/sdk/io/jdbc/JdbcIO.java

```diff
@@ -1432,6 +1442,11 @@ void set(
       return toBuilder().setPreparedStatementSetter(setter).build();
     }
 
+    /** If true, enables using a dynamically determined number of shards to write. */
+    public WriteWithResults<T, V> withAutoSharding() {
```

Comment:
Should we allow to customise a sharding function?

Reply:
hmmm I think we should wait for feedback before adding that capability. If users find that they want more control over sharding, they'd probably file issues / ask questions on stackoverflow. If we add this capability we'll need to support it for the longer term... What do you think?

Reply:
Yeah, that sounds like a separate feature. The purpose of this feature is for the runner to determine the appropriate amount of sharding automatically.

Reply:
Ok, let's keep it only automatically for now.